src/stdLambda.cls

```diff
@@ -747,11 +764,143 @@ Private Sub parseFlowPriority1()
+        Dim SkipLoop As Integer : SkipLoop = addOperation(oJump, ifTrue, , -1)  ' Skip loop
+        Call parseBlock("next")                                                 ' Get everything in the loop
+        Call ForLoopIncrement(varName, Operator, OperatorValue, Increment)      ' Create Increment line index = index + x
+        Call addOperation(oJump, , RepeatLoop, -1)                              ' Goes back to loop
+        This.operations(SkipLoop).value = This.iOperationIndex                  ' Set jump position for skip-condition
     Else
         Call parseValuePriority1
     End If
 End Sub
 
+' the user might define the loop as:
+' for index = 1 to 10
+' problem here is, that 1 needs to be assigned to index 
+' this sub just inserts the "let" token before index
+Private Sub ForLoopLet()
+    If peek("let") = False Then
+        Dim NewToken As token
+        NewToken.value = "let"
+        NewToken.Type = getTokenDefinitionByName(getTokenDefinitions(), "let")
+        Call InsertToken(this.Tokens, NewToken, this.iTokenIndex)
+    End If
+    Call parseBlock("to")
+    Call consume("to")
+End Sub
+
+' for index 1 to 10 usually means loop until index > 10
+' this sub gets rid of the "to" token and replaces it with the operator >
+' this sub also sets up the break condition
+Private Sub ForLoopCondition(varName As String)
+    Dim NewToken As token
+    Dim defs() As TokenDefinition
+    defs = getTokenDefinitions
+
+    NewToken.value = varName
+    NewToken.Type = getTokenDefinitionByName(defs, "var")
+    Call InsertToken(this.Tokens, NewToken, this.iTokenIndex - 1)
+
+    NewToken.value = ">"
+    NewToken.Type = getTokenDefinitionByName(defs, "greaterThan")
+    Call InsertToken(this.Tokens, NewToken, this.iTokenIndex)
+
+    this.Tokens = removeToken(this.Tokens, this.iTokenIndex + 1)
+    this.iTokenIndex = this.iTokenIndex - 1
+End Sub
+
+' for index 1 to 10 step +2
+' the user might not use step..., this sub will create it
+' if it is defined it will change the condition and the final index-incrementation
+' Operator is the token-name of the increment operator --> if step +2 then operator = add | if step -2 then operator = subtract
+' OperatorValue is the actually ascii character for the operator --< +|-|*|/|^
+' Increment is the value by which index will be increased/decreased
+' These 3 variables come from outside and are used for ForLoopIncrement
+Private Sub ForLoopStep(ByRef Operator As String, ByRef OperatorValue As String, ByRef Increment As String)
+    Dim defs()       As TokenDefinition
+    Dim Condition    As TokenDefinition
+    Dim NewToken     As token
+    defs = getTokenDefinitions()
+    If peek("step", 4) Then
+        Select Case True
+            Case peek("add"     , 5) : Operator = "add"      : OperatorValue = "+": Increment = this.Tokens(this.iTokenIndex + 5).value: Condition = getTokenDefinitionByName(defs, "greaterThan")

```

Comment:
in my mind this should be an optConsume too. And we should parse a regular value/expression directly afterwards.

e.g. `for i = 1 to 10 step 11+32*someFunc()`